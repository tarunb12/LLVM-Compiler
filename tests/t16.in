// Function test (prime factorization)

int primeFactors(int k) {
    int n = k
    while (n % 2 == 0) {
        printf ("%d ", 2)
        n = n / 2
    }
    int i
    for (i = 3; i < n / 2; i += 2) {
        while (n % i == 0) {
            printf("%d ", i)
            n = n / i
        }
    }
    if (n > 2) printf("%d ", n)
    return 0
}

int main() {
    int n = 436
    printf("Prime factors of %d: ", n)
    primeFactors(n)
    return 0
}