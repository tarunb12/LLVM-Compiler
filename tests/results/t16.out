Prime factors of 436: 2 2 109 